### PR TITLE
javascript/leap: round 2 note improvements

### PR DESCRIPTION
**Pull request by SleeplessByte**
Improves the notes with:
- templates to copy and paste
- instructions on when to approve and when to only suggest
- more suggestions around short-circtuiting, ordering and parentheses uses

All these changes are based on 200+ mentoring sessions.

@exercism/website-copy-javascript 
@exercism/website-copy-reviewers  I'm looking for a ✔️ review -- these changes should be merged soon to helpt our mentors with the 300+ `leap` submissions. I'll make subsequent edits later. However, please spell-check, or suggest wording changes ❤️ .

### CODE REVIEW

File: tracks/javascript/exercises/leap/mentoring.md [hunk 1]
```diff
@@ -5,71 +5,186 @@ The point of this exercise is to teach the student to use:
...
+export const isLeap = year => isDivisibleBy4(year) && (!isDivisibleBy100(year) || isDivisibleBy400(year))
+```
 
-  export const isLeap = year => isDivisibleBy4(year) && (!isDivisibleBy100(year) || isDivisibleBy400(year))
-  ```
+The particulary clever student will use `Date`. Do not punish them, for this, and simply appreciate the ingenuity:
+```javascript
+// Months in Date start at 0, this tries to set the date to 29th of february and
+// checks if it has rolled over to March.
+export const isLeap = (year) => new Date(year, 1, 29).getMonth() === 1
```
Comment (tejasbubane) [inline tracks/javascript/exercises/leap/mentoring.md]: Wow! I did not know this!

File: tracks/javascript/exercises/leap/mentoring.md [hunk 2]
```diff
@@ -5,71 +5,186 @@ The point of this exercise is to teach the student to use:
...
+      - `b` is _only_ evaluated if `b` was `falsy`, otherwise the entire expression short-circuits to whatever `a` yielded.
+
+    In the wild you'll find code like this:
+    ```javascript
+    // only execute functionname if variable is truthy,
+    //   usefull for flags, or preconditions
+    variable && functionname()
+
+    // only execute functionname if variable is falsy,
+    //   usefull for flags, or defaults, or error cases
```
Comment (tejasbubane) [inline tracks/javascript/exercises/leap/mentoring.md]: typo `useful`

File: tracks/javascript/exercises/leap/mentoring.md [hunk 3]
```diff
@@ -5,71 +5,186 @@ The point of this exercise is to teach the student to use:
...
+Here is a template you can respond with:
+
+    That is a good question. In JavaScript and Typescript, some operators have [short-circuit](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Short-circuit_evaluation) evaluation.
+
+    - `a && b`
+      - `a` is _always_ evaluated
+      - `b` is _only_ evaluated if `a` was `truthy`, otherwise the entire expression short-circuits to whatever `a` yielded.
+    - `a || b`
+      - `a` is _always_ evaluated
+      - `b` is _only_ evaluated if `b` was `falsy`, otherwise the entire expression short-circuits to whatever `a` yielded.
```
Comment (tejasbubane) [inline tracks/javascript/exercises/leap/mentoring.md]: `yielded` might sound slightly confusing, maybe use `evaluated`? Although both are same, ignore if you feel this is OK.

**Comment by tejasbubane** [comment]
This is extermely nicely written @SleeplessByte! Just a couple of comments and we should be good.

**Comment by SleeplessByte** [comment]
Done and done @tejasbubane

**Pull request merged**